lib/events/event_handler.ex

```diff
@@ -0,0 +1,79 @@
+defmodule ElixirTools.Events.EventHandler do
+  alias ElixirTools.Events.{Event, NotSentEvent}
+
+  @type events_opt ::
+          {:event_handler_module, module}
+          | {:task_supervisor_module, module}
+          | {:event_module, module}
+          | {:not_sent_event_module, module}
+          | {:telemetry_module, module}
+  @callback send_event(any, [events_opt]) :: :ok
+
+  @callback create(event_name, payload) :: Event.t()
+  @callback publish(Event.t(), [events_opt]) :: :ok
+
+  @optional_callbacks send_event: 2,
+                      create: 2,
+                      publish: 2
+
+  @typep event_name :: String.t()
+  @typep payload :: map
+  @typep event_id_seed :: Ecto.UUID.t()
+  @typep event_id_seed_optional :: String.t()
+
+  @spec create(event_name, payload, event_id_seed) :: Event.t()
+  def create(event_name, payload, event_id_seed) do
+    %Event{name: event_name, payload: payload, event_id_seed: event_id_seed}
+  end
+
+  @typep create_optional ::
+           {:event_id_seed_optional, event_id_seed_optional}
+           | {:occurred_at, DateTime.t()}
+  @spec create(event_name, payload, event_id_seed, [create_optional]) :: Event.t()
+  def create(event_name, payload, event_id_seed, create_optional) do
+    event_id_seed_optional = create_optional[:event_id_seed_optional] || ""
+    occurred_at = create_optional[:occurred_at]
+
+    %Event{
+      name: event_name,
+      payload: payload,
+      event_id_seed: event_id_seed,
+      event_id_seed_optional: event_id_seed_optional,
+      occurred_at: occurred_at
+    }
+  end
+
+  @spec publish(Event.t(), [events_opt]) :: :ok
+  def publish(event, opts) do
+    task_supervisor_module = opts[:task_supervisor_module] || Task.Supervisor
+
+    task_supervisor_module.async_nolink(ElixirTools.TaskSupervisor, fn ->
+      publish_event_call(event, opts)
```

Comment:
it's a bit tricky to remove this function because of this https://github.com/digitalorigin/pg-elixir-tools/pull/43#discussion_r317036685. So it's better to do in another PR, I would say